### ISSUE TITLE
Malta settings correct?

### ISSUE THREAD

**Issue by bahnjee** [MicrosoftDocs/windows-itpro-docs#5415]
In the very last screenshot (Malta Data Source config), you show that it should be configured to connect using the MBAM Read Only account.  I've been through setup several times, checking my entries very carefully (because things aren't working) and each time I end up with my Read/Write account showing up here.  The "Test Connection" succeeds also.  (I've not tested with the Read only account entered here because our man that has the ReadOnly account password is not available.)

While I'm at it...  the error I'm receiving is that shown below and if you could point to a resource for troubleshooting that, I would be forever in your debt.

Error in the SelfService website tracelog:
---------------------------------------------
QueryRecoveryKeyIdsForUser: an error occurred while logging an audit event to the compliance database. Message: Procedure or function AddRecoveryAuditEvent has too many arguments specified..

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5c6c4323-445a-3ff1-5ff0-76513e66bc92
* Version Independent ID: 9df4765e-206e-2c24-8b4b-1bfb762a18ea
* Content: [Troubleshooting MBAM 2.5 installation problems](https://docs.microsoft.com/en-ca/microsoft-desktop-optimization-pack/mbam-v25/troubleshooting-mbam-installation)
* Content Source: [mdop/mbam-v25/troubleshooting-mbam-installation.md](https://github.com/MicrosoftDocs/windows-itpro-docs/blob/public/mdop/mbam-v25/troubleshooting-mbam-installation.md)
* Product: **w10**
* Technology: **windows**
* GitHub Login: @Delhan
* Microsoft Alias: **delhan**

**Comment by bahnjee** [comment]
...if it matters, I'm setting this up with SCCM integration...

**Comment by mypil** [comment]
@bahnjee   - Thank you for submitting feedback.

I think the best way forward is if you can contact support and open a **[service ticket](https://support.microsoft.com/hub/4338813/windows-help?os=windows-10)** for Windows 10 products  so this can get resolved ASAP. Based on outcome, let me know if it is something that can be called out in the docs.

If you don't mind please keep us posted here on the resolution and I really want to get this information you are discovering back into the docs.

**Comment by bahnjee** [comment]
I finally got it working (and the screenshot I referenced in my initial post is correct as published).  Turned out that my colleague who started the upgrade to MBAM 2.5 SP1 didn't uninstall MBAM from the DB server before he installed the update (and then foisted the project off on me).  As soon as I did the uninstall/reinstall and re-added the roles, the "too many arguments" error went away and all is well in my MBAM world now.

I recognize that MBAM is a complex system with a lot of moving parts, but I don't think I'm alone in thinking the documentation could be clearer.  In my searching to find answers to my  troubleshooting, I saw a great many posts wherein others were equally lost considering the correct setup.  An example is the differences between choosing a user or a group for the various roles.  Some notes on pros and cons of users vs. groups would have been useful.  Similarly, details on how to choose the number of servers and which server should host which roles should be provided.

It's not my intent to bash anyone, and some of this long missive is no doubt a result of my frustration towards the aforementioned colleague.  I realize documenting complex systems can be difficult.  But those tasked with documenting should step back and look at the docs they're writing and view them from the point of view of their audience.  If I'm reading the docs, then I have much less knowledge than the writers and would benefit from clear explanations and examples of configurations.  A "Best Practices" guide would be beneficial, too.
